cloud/wfm/src/task/BaseTask.ts

```diff
@@ -0,0 +1,46 @@
+import { EventEmitter } from 'eventemitter3';
+import { Task, TaskEventData, TaskStatus } from './index';
+
+/**
+ * Base implementation for
+ */
+class BaseTask extends EventEmitter implements Task {
+  protected _status: TaskStatus | number = TaskStatus.pending;
+  protected options: object;
+
+  public set status(to: TaskStatus | number) {
+    const e: TaskEventData<this> = {
+      date: new Date(),
```

Comment:
` date: new Date(),` you want to track all status changes here? How often it can be tracked, otherwise it can lead to memory leaks